### PR TITLE
Ports TauCeti's welding effect

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

Port of: https://github.com/TauCetiStation/TauCetiClassic/pull/6540 with heavy changes to support our... ways.

It telegraphs to the player what exactly is being welded. The effect is a bonus.

https://user-images.githubusercontent.com/9026500/156419204-e8ff0549-1188-42e7-82dd-a11a41d4f4d8.mp4

## Changelog

:cl:
add: Welding now displays a spark effect.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/welding_sparks/code/weldingtool.dm [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * These are the side effects of our choice to achieve salvation.
+ */
+
+/obj/item
+	/// An image effect for displaying a welding effect for items that support it.
+	var/image/welding_sparks
+
+/obj/item/Initialize(mapload)
+	. = ..()
+	if(tool_behaviour == TOOL_WELDER) // If we act like a welder, set up some spark effects.
```
Comment (Zonespace27) [inline modular_skyrat/modules/welding_sparks/code/weldingtool.dm:17]: early return

Comment (Gandalf2k15) [reply modular_skyrat/modules/welding_sparks/code/weldingtool.dm:17]: No, not on initialize.

Comment (Zonespace27) [reply modular_skyrat/modules/welding_sparks/code/weldingtool.dm:17]: '. = ..()' makes the default value on a 'return' with no arguments whatever '..()' returned, so no harm in doing that

Comment (Gandalf2k15) [reply modular_skyrat/modules/welding_sparks/code/weldingtool.dm:17]: it makes little difference but it's best to follow guidelines

**Pull request merged**